### ISSUE TITLE
ClusterShell timers are too fast

### ISSUE THREAD

**Issue by cedeyn** [cea-hpc/clustershell#399]
Playing with ClusterShell timer and EventHandler, I hit this bug :

```
#!/usr/bin/python
import sys                                                                                             
import time
from ClusterShell import Task                                                                          
from ClusterShell import Event

delay = 0.2
timeout = 0.1                                                                                          
loop = 3

class TestHandler(Event.EventHandler):
    def ev_timer(self, timer):                                                                         
        sys.stdout.write("Delay fired\n")                                                              
        task.shell("/bin/sleep 5", timeout = timeout, handler = TestHandler())                         
task = Task.task_self()                                                                                
start = time.time()

for i in range(loop):                                                                                  
    task.timer(handler=TestHandler(), fire=delay)                                                      
    task.run()
    task.join()                                                                                        
stop = time.time()
duration = stop - start      
sys.stdout.write("Process duration = {0} shoud be at least {1}\n".format(duration, (delay + timeout) * loop))
```
The total duration of this code is significantly lower than expected...
It seems the ev_timer is fired too quickly.

**Comment by cedeyn** [comment]
It seems to be only reproduced on VM:
```
[@vm0 ~]$ python /tmp/repro.py 
Delay fired
Delay fired
Delay fired
Process duration = 0.899127960205 shoud be at least 0.9

[@baremetal]$ python /dev/shm/repro.clustershell_timer.py 
Delay fired
Delay fired
Delay fired
Process duration = 0.903531074524 shoud be at least 0.9

```

**Comment by cedeyn** [comment]
More informations:
On vm: Python 2.7.15
On baremetal: Python 2.7.5

**Comment by thiell** [comment]
Hey @cedeyn, could you try https://review.gerrithub.io/c/cea-hpc/clustershell/+/452363 and let me know how it goes? (because I am not able to reproduce the issue myself, my VM is not as fast :)), but I think I found a possible cause (introduced when we fixed #254)

Before this patch, we "fired" all timers that had expired + the ones that were just about to expire (all timers with a firedate up to now+EPSILON). With this patch, we "fire" only expired timers. That makes actually more sense and should fix your issue. But, the previous now+EPSILON thing had a side effect: a tendency to "batch" close-in-time timers. This is why I also had to fix a test.

**Comment by cedeyn** [comment]
Seems to fix the issue :)
Before the patch, I hit the bug after 75 loops.
With the patch and after 1000 loops everything goes well.

Good catch !
